blocks/library/paragraph/index.js

```diff
@@ -314,16 +356,28 @@ export const settings = {
 	edit: ParagraphBlock,
 
 	save( { attributes } ) {
-		const { width, align, content, dropCap, backgroundColor, textColor, fontSize } = attributes;
+		const {
+			width,
+			align,
+			content,
+			dropCap,
+			backgroundColor,
+			textColor,
+			fontSize,
+		} = attributes;
+
+		const thresholdFontSize = findKey( FONT_SIZES, ( size ) => size === fontSize );
+
 		const className = classnames( {
 			[ `align${ width }` ]: width,
 			'has-background': backgroundColor,
 			'has-drop-cap': dropCap,
-		} );
+		}, thresholdFontSize ? `is-${ thresholdFontSize }-text` : undefined );
+
 		const styles = {
 			backgroundColor: backgroundColor,
 			color: textColor,
-			fontSize: fontSize,
+			fontSize: thresholdFontSize ? null : fontSize,
```

Comment:
Normally in this situation, I would use thresholdFontSize ? undefined : fontSize,. But in this case, I think using null is safe as false values will be ignored.